src/Components/WebView/WebView/src/WebViewManager.cs

```diff
@@ -166,6 +166,37 @@ protected void MessageReceived(Uri sourceUri, string message)
         });
     }
 
+    /// <summary>
+    /// Calls the specified <paramref name="workItem"/> asynchronously and passes in the scoped services available to Razor components.
+    /// </summary>
+    /// <param name="workItem">The action to call.</param>
+    /// <returns>Returns a <see cref="Task"/> representing <c>true</c> if the <paramref name="workItem"/> was called, or <c>false</c> if it was not called because Blazor is not currently running.</returns>
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="workItem"/> is <c>null</c>.</exception>
+    public async Task<bool> TryDispatchAsync(Action<IServiceProvider> workItem)
+    {
+        ArgumentNullException.ThrowIfNull(workItem);
+
+        if (_currentPageContext is null)
+        {
+            return false;
+        }
+
+        var capturedCurrentPageContext = _currentPageContext;
+
+        return await _currentPageContext.Renderer.Dispatcher.InvokeAsync(() =>
+        {
+            if (capturedCurrentPageContext != _currentPageContext)
```

Comment:
@SteveSandersonMS - does this match what you were thinking? Check consistency of captured context, and if it doesn't match, "fail" (by returning `false`).

Reply:
Yes exactly.